BinaryEncoding.md

```diff
@@ -439,28 +439,25 @@ It is legal to have several entries with the same type.
 | Name | Opcode | Immediates | Description |
 | ---- | ---- | ---- | ---- |
 | `unreachable` | `0x00` | | trap immediately |
-| `block` | `0x01` |  | begin a sequence of expressions, the last of which yields a value |
-| `loop` | `0x02` |  | begin a block which can also form control flow loops |
-| `if` | `0x03` | | begin if expression |
+| `block` | `0x01` | arity : `varuint1` | begin a sequence of expressions, yielding 0 or 1 values |
+| `loop` | `0x02` | arity : `varuint1` | begin a block which can form control flow loops |
```

Comment:
`loop` having an arity is awkward, because branches targeting a `loop` label are effectively required to have arity 0. A common validation strategy would record an arity of 0 in the control-flow stack entry for a `loop`, and then need to separately record the `loop` arity somewhere else.

Restricting `loop` to arity 0 would simplify it, and obviate its arity immediate. `loop`s are less frequent than `block`s, so the code size win from using `loop`'s return value(s) directly are likely to be small, especially after accounting for the arity immediates it would add to every `loop`.